src/components/table/ProfileEditView.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { FC } from 'react';
+import { EditUserType } from 'types/ListTypes';
+import TextField from '@mui/material/TextField';
+import { Button } from '@mui/material';
+import * as yup from "yup"
+import { yupResolver } from "@hookform/resolvers/yup"
+import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+import { useAuthContext } from 'context/AuthContext';
+
+
+type EditViewProps = {
+    user: EditUserType;
+    setIsEditing: (value: boolean) => void;
+
+}
+
+type Inputs = {
+    email: string,
+    newEmail: string,
+    password: string,
+    newPassword: string,
+}
+
+const schema = yup.object({
+    email: yup.string().required(),
+    newEmail: yup.string().required(),
+    password: yup.string().min(8).required(),
+    newPassword: yup.string().min(8).required(),
+
+})
+
+const ProfileEditView: FC<EditViewProps> = ({ user, setIsEditing }) => {
+    const { updateClient } = useAuthContext();
+    const navigate = useNavigate();
+
+    const { register, formState: { errors }, handleSubmit } = useForm<Inputs>({
+        resolver: yupResolver(schema),
+        defaultValues: { password: user.password, email: user.email }
+    });
+
+
+    const onSubmit = (values: Inputs) => {
+        const editedUser = { ...user, password: values.newPassword, email: values.newEmail }
```

Comment:
Zmiana hasla to powininen byc oddzielny komponent- tak jak to spotykasz na roznych stronach. Podajesz stare i wpisujesz nowe. Na razie okay, ale mozesz dorobic wspomiany.